basic-teal/.Rprofile

```diff
@@ -1 +1,7 @@
 source("renv/activate.R")
+
+for (package in renv::dependencies()$Package) {
+    if (!requireNamespace(package)) {
+        renv::install(package, prompt = FALSE)
+    }
+}
```

Comment:
I am not super convinced to put installs into the .Rprofile. Any install action should be called explicitly for convenience and clarity. We should run additional install alongside restore()

Reply:
WDYT about adding some helpful comments?
Say: _"Installing the latest versions of teal packages"_

With an additional modification to the logic by installing the intersection of `available.packages(repo = "https://insightsengineering.r-universe.dev")` and `renv::dependencies()$Package` just like the CI?

Reply:
Well it's not necessarily about the communication to the user but rather some non-standard way of having this in a `.Rprofile` file which is a very special file. If an user will open R shell from within that dir then he will get interactivity back after all the packages will be installed. That's some bad experience. Usually, any installation is performed on explicit install calls (or any wrapper around that such as restore()) that I am missing here.
Please also see my comment here: https://github.com/insightsengineering/teal.gallery/pull/41#discussion_r1297510377